src/server/DetailsError.ts

```diff
@@ -1,15 +1,28 @@
-class DetailsError extends Error {
+export default class DetailsError extends Error {
+
+  /**
+   * HTTP response status code by the server to the client
+   */
   statusCode: number;
+
+  /**
+   * Description to show in the client
+   */
   description: string;
-  details: string;
 
-  constructor(message: string, status: number, description: string, details: string) {
-    super(message);
+  /**
+   * Details about the info that can help to fix
+   */
+  details: string;
 
-    this.statusCode = status;
+  constructor(name: string, message: string, code: number, description: string, details: string) {
+    super();
+    this.name = name;
```

Comment:
this can help to reduce the code (and protect the properties to be mutated later)

```javascript 
constructor(readonly paramName: paramType)
```